### PR TITLE
[13.0] [IMP] github_connector: Added the option to authenticate with tokens

### PR DESCRIPTION
**Pull request by CarlosRoca13**
cc @Tecnativa TT24646

please @pedrobaeza @Yajo, can you review it?

### CODE REVIEW

**Review by yajo — APPROVED** [review]
Basically the same as https://github.com/OCA/interface-github/pull/54#pullrequestreview-454031591.

**Comment by pedrobaeza** [comment]
Follow 12.0 changes. Next time, you can see that is better to propose only one version, and when finished, then start with the next one.

**Comment by CarlosRoca13** [comment]
@pedrobaeza Changes done!

File: github_connector/models/github.py [hunk 1]
```diff
@@ -113,15 +112,15 @@ def get_by_url(self, url, call_type, data=False):
                 )
                 % self.login
             )
-        elif response.status_code == _CODE_403:
+        if response.status_code == _CODE_403:
```
Comment (pedrobaeza) [inline github_connector/models/github.py]: Why removing the `elif`?

File: github_connector/README.rst [hunk 2]
```diff
@@ -45,13 +45,20 @@ Once installed, you have to:
 #. Open your odoo.conf file and add extra settings to mention Github
    credentials, and the local path where the source code will be downloaded:
 
-   * ``source_code_local_path = /workspace/source_code/``
+    * ``source_code_local_path = /workspace/source_code/``
 
 Note: make sure that Odoo process has read / write access on that folder
 
+    * ``github_token = your_github_access_token``
```
Comment (pedrobaeza) [inline github_connector/README.rst]: This seems to be not properly indented

**Comment by CarlosRoca13** [comment]
@pedrobaeza changes done

**Comment by pedrobaeza** [comment]
Ready?

**Comment by CarlosRoca13** [comment]
@pedrobaeza Yes

**Comment by CarlosRoca13** [comment]
Wait, I found one bug on branch sync

**Comment by CarlosRoca13** [comment]
Okey, that was not a bug, just my fault, it is ready to be merged

**Review by pedrobaeza — APPROVED** [review]

**Pull request merged**